src/transformers/image_utils.py

```diff
@@ -62,6 +62,159 @@
+        return _cast_to_image_object(result)
+
+    # If the output is a method, we wrap it to cast the output to an `ImageArray` if it is a NumPy array.
+    if callable(output) and not isinstance(output, type):
+        return _method_decorator
+
+    return _cast_to_image_object(output)
+
+
+class ImageArray(np.lib.mixins.NDArrayOperatorsMixin):
+    """
+    Array container for images. This class is a wrapper around `np.ndarray` that includes image specific information
+    such as the channel dimension format and the number of channels.
+
+    See: https://numpy.org/doc/stable/user/basics.dispatch.html
+
+    Args:
+        data (`ImageInput`):
+            The image data.
+        data_format (`ChannelDimension`, *optional*):
+            The channel dimension format of the image. If `None`, will be inferred from the image data.
+        num_channels (`int`, *optional*):
+            The number of channels in the image. If `None`, will be inferred from the image data.
+    """
+
+    def __init__(
+        self, data: ImageInput, data_format: Optional[ChannelDimension] = None, num_channels: int = None
+    ) -> None:
+        if isinstance(data, ImageArray):
+            data = data._data
+            data_format = data._data_format
+            num_channels = data._num_channels
+
+        if isinstance(data, PIL.Image.Image):
+            if data.mode in ("1", "L", "P", "I;16", "I", "F"):
+                data_format = ChannelDimension.NONE
+            else:
+                # PIL images are always channels last when converted to numpy arrays
+                data_format = ChannelDimension.LAST
+
+        data = np.array(data)
+
+        if data_format is None:
+            data_format = infer_channel_dimension_format(data)
+
+        if data_format == ChannelDimension.NONE:
+            num_channels = 0
+            height = data.shape[-2]
+            width = data.shape[-1]
+        elif data_format == ChannelDimension.FIRST:
+            num_channels = data.shape[-3]
+            height = data.shape[-2]
+            width = data.shape[-1]
+        elif data_format == ChannelDimension.LAST:
+            num_channels = data.shape[-1]
+            height = data.shape[-3]
+            width = data.shape[-2]
+        else:
+            raise ValueError("Invalid channel dimension format")
+
+        self._data = data
+        self._data_format = data_format
+        self._num_channels = num_channels
+        self._height = height
+        self._width = width
+
+    def __getattribute__(self, __name: str) -> Any:
+        if __name in ("_data",):
+            return super().__getattribute__(__name)
+
+        if hasattr(self._data, __name):
+            result = getattr(self._data, __name)
+        else:
+            result = super().__getattribute__(__name)
+        return _output_wrapper(result)
+
+    def __getitem__(self, key: Any) -> Any:
+        result = self._data[key]
+        return _output_wrapper(result)
```

Comment:
This will fail when we want to fetch a 1D sub-array:

```python
import numpy as np
from transformers.image_utils import ImageArray

np_data = np.ones(shape=(3, 224, 224))
img_array = ImageArray(np_data)

# This fails, as it is 1-D and can't be converted to `ImageArray`
img_array[0, 0, 0]
```

Reply:
Same comment also applies to `__getattribute__`

Reply:
I've added a check to see if we can create an ImageArray from the numpy array and this as a test :) https://github.com/huggingface/transformers/pull/24577/commits/01cbfcb955cc6d648a3666fbf763f363681ef6b0